src/MinimalDelegation.sol

```diff
@@ -35,4 +48,20 @@ contract MinimalDelegation {
         delete MinimalDelegationStorageLib.get().keyStorage[keyHash];
         emit Revoked(keyHash);
     }
+
+    function supportsExecutionMode(bytes32 mode) external pure override returns (bool result) {
+        return mode.isBatchedCall();
+    }
+
+    function _authorizeCaller() private view {
+        if (msg.sender != address(this)) revert IERC7821.Unauthorized();
+    }
+
```

Comment:
nit: can we add a comment that this technically allows for unrestricted self calls right now, and a TODO is to add authentication on a call target basis